# 62656827 Fixes bug that allows provision of flight dates earlier than check in date

SrMoffat opened the pull request:
## What does this PR do?
Fixes the bug that allows a user to provide flight dates that are earlier than the user's check in date in the destination guest house 

## Description of Task to be completed?
Currently, when uploading `checklist items` when a user provide flight dates that are `earlier than` the planned `check-in date`, the transaction is completed successfully. This creates a loop hole that allows the user arrive earlier than planned, leaving the traveller stranded. This PR aims at solving this problem.

## How should this be manually tested?
- Clone the repo
    `git clone https://github.com/andela/travel_tool_front.git`
- Switch to the project directory
   `cd travel_tool_front`
- Checkout the **bg-inconsistent-flight-dates-162656827** branch
   `git checkout bg-inconsistent-flight-dates-162656827`
- Install dependencies and start the server
   - `yarn install` then `yarn start` *if you do not have docker*
   - `make start` *if you have docker*
 - Start the [backend](https://github.com/andela/travel_tool_back) of this project.
 - Navigate to [travela](http://travela-local.andela.com:3000/) in your browser and log in with your Andela account. 
- Navigate to [travel requests](http://travela-local.andela.com:3000/requests) in your browser.
- Create a new request and get the request approved
- Ensure the travel dates selected have available rooms/bed spaces at the destination centre
- While planning the trip, click on the `request menu` (three dots on the extreme right)
- Select the `Travel Checklist` option

**Expected**
- The user should not be able to provide flight dates that are earlier than the planned check in date (departure time in the request). The invalid dates are `dark grey` and `inactive` in Chrome but the valid ones are `white` and `active` (permissible date range)

## Any background context you want to provide?
N/A

## What are the relevant pivotal tracker stories?
#162656827
Story is [here](https://www.pivotaltracker.com/story/show/1162656827)

## Screenshots (if appropriate)

N/A
## Questions:
N/A

amarachukwu-agbo commented:
@SrMoffat , if the default dates that are pre-populated by the system are left as-is and the user fills in the other fields, the ticket details are not uploaded automatically. Please look into this.

Also, look into some of the code climate issues that can be resolved as well as the drop in test coverage.

For the PR description, please fix your story ID in the PR title. In the section  for the PT story, providing the story ID as a link to the PT board is sufficient

SrMoffat commented:
> if the default dates that are pre-populated by the system are left as-is and the user fills in the other fields, the ticket details are not uploaded automatically. Please look into this.

Hi @amarachukwu-agbo ,
I have done the following as advised:
- Updated the `PR title` from `#62656827` to `#162656827 `
- Updated `PR description` with the `PT storyID as a link` and removed unnecessary information
- Fixed the sensible code climate issues, the current ones either are not in my code or don’t make sense e.g. prop validation as repetition

amarachukwu-agbo commented:
@SrMoffat , we can have a sync on this
